### PR TITLE
Trim whitespace before make upload

### PR DESCRIPTION
**Pull request by rumyantseva**

### CODE REVIEW

**Review by AlekSi — CHANGES_REQUESTED** [review]
I think that's a mistake to do `rm -fr` in `upload` target. I propose to move installation part to `init`; `whitespace` target should only trim whitespace. Also, trimmed whitespace should be committed; it is not enough to trim it before the rsync.

**Comment by AlekSi** [comment]
We can also move `strip-whitespace.go` to that repository and `go run` it. I think that would be even better.

**Comment by AlekSi** [comment]
Also please remove submodule accidentally added in the previous PR.

**Comment by rumyantseva** [comment]
> Also please remove submodule accidentally added in the previous PR.

@AlekSi sorry, I don't understand this part. What submodule?

**Comment by AlekSi** [comment]
https://github.com/GolangShow/golangshow.com

**Comment by rumyantseva** [comment]
Fixed!

**Comment by rumyantseva** [comment]
Fixed it again.

File: Makefile [hunk 1]
```diff
@@ -1,23 +1,28 @@
...
+html:
 	rm -fr public
 	hugo --theme=golangshow
 
+.PHONY: whitespace
+whitespace:
+	go run tools/strip-whitespace/main.go
+
+.PHONY: build
+build: html whitespace
```
Comment (AlekSi) [inline Makefile]: Why `build`, `html` and `whitespace` are separate targets? In what case we need them separately?

File: Makefile [hunk 2]
```diff
@@ -1,23 +1,28 @@
...
+	$(info -------------------------------------------------------------------------------------)
+	$(info Yeah! The code is ready for upload, please don't forget to check, commit and push it.)
+	$(info After that, you can run `make upload` to publish the site.)
+
+.PHONY: watch
```
Comment (AlekSi) [inline Makefile]: We can collect all `.PHONY` targets at the end like
```
.PHONY: watch init …
```

However, I only see a benefit in them if we have make targets sharing names with file or directory names. We don't have a `watch` directory, for example, so why we need a phony target at all?

File: Makefile [hunk 3]
```diff
@@ -1,23 +1,28 @@
...
 
+.PHONY: init
 init:
 	brew update
 	brew install hugo
 
-upload: build
-	git push
+.PHONY: upload
+upload:
```
Comment (AlekSi) [inline Makefile]:
```sh
git diff --exit-status
git push
```

I don't like `git push` there, but it was required in practice.

**Review by AlekSi — CHANGES_REQUESTED** [review]

File: Makefile (hunk 3 above)
Comment (AlekSi) [reply Makefile]: Err, `git diff --exit-code`.

**Pull request merged**